### PR TITLE
Arcs logic into cdn

### PR DESCRIPTION
**Pull request by smalls**
Overall this moves the bulk of the arcs logic into the arcs-cdn repo (next to the arcs code). There are parallel changes to move logic out of arcs in that repository.

### CODE REVIEW

File: dev/apps/chrome-extension/index.html [hunk 1]
```diff
@@ -45,25 +45,91 @@
...
-        // always take the browser arc public
-        StorageTools.saveProfileState(true);
-      }
+        var appshell = this;
+
+        window.addEventListener('message', event => {
+          if (event.source != window || event.data.method != 'injectArcsData') {
+            return;
+          }
+          console.log('do arcs!1!', event.data.entities);
```
Comment (shaper) [inline dev/apps/chrome-extension/index.html:65]: Do we want to leave in this and next line?

File: dev/apps/chrome-extension/index.html [hunk 2]
```diff
@@ -45,25 +45,91 @@
...
+          console.log(config);
+
+          // reduce the deeply nested structure of url=>entities-of-many-types
+          // to a flatter, combined form of type=>entities.
+          let dataByType = Object.entries(event.data.entities).reduce( (accumulator, [key, value]) => {
+            console.log('a', key, value);
```
Comment (shaper) [inline dev/apps/chrome-extension/index.html:71]: Remove?

File: v0.2.1/app-shell/app-shell.html [hunk 3]
```diff
@@ -360,6 +360,7 @@
       let manifests = await this._fetchManifestList();
       return await ManifestTools.loadManifest(this._config, this._loader, manifests);
     }
+
```
Comment (shaper) [inline v0.2.1/app-shell/app-shell.html:363]: Do we need to include the v0.2.1 change here and below for the extension?

**Review by shaper — APPROVED** [review]

**Comment by smalls** [comment]
So.. sorry. This change ended up being confusing and I forgot to mention this.

I was developing in dev, but when Scott moved app-shell.2 over to dev I bounced out to v0.2.1. The version that was left in dev is a bit stale (and unworking due to app-shell.2 changes).

Once we have a chance to try this out and decide that it's stable (and doesn't need any other features) I'll update the version in dev - or, wait until there's another stable release.

File: v0.2.1/app-shell/app-shell.html (hunk 3 above)
Comment (smalls) [reply v0.2.1/app-shell/app-shell.html:363]: I'm not sure I understand. Does my [comment](https://github.com/PolymerLabs/arcs-cdn/pull/46#issuecomment-347344510) answer this at all?

Comment (shaper) [reply v0.2.1/app-shell/app-shell.html:363]: Yes, your comment answers, thanks.

**Pull request merged**